lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,471 @@
+
+      self.test_url_verify = 'Contact priority for url'
+      self.live_url_verify = 'Contact priority for url'
+
+      self.test_url_batch = 'https://qa.api.mxmerchant.com/checkout/v3/batch'
+      self.live_url_batch = 'https://qa.api.mxmerchant.com/checkout/v3/batch'
+
+      self.test_auth = 'Contact priority for token'
+      self.live_auth = 'Contact priority for token'
+      self.test_env_verify = 'qa-aus'
+      self.live_env_verify = 'qa-aus'
+
+      require 'byebug'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'http://www.example.net/'
+      self.display_name = 'Priority'
+      # byebug
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        # requires!(options, :key, :secret)
+        # @key, @secret = options.values_at(:key, :secret)
+        super
+      end
+
+      def basic_auth(key, secret)
+        Base64.strict_encode64("#{key}:#{secret}")
+      end
+
+      def request_headers(options)
+        {
+          'Content-Type' => 'application/json',
+          'Authorization' => "Basic #{basic_auth(options[:key], options[:secret])}"
+        }
+      end
+
+      def request_verify_headers
+        auth = test? ? self.test_auth : self.live_auth
+        env_verify = test? ? self.test_env_verify : self.live_env_verify
+        {
+          'Authorization' => auth.to_s,
+          'EOS-Virtual-Environment' => env_verify.to_s
+        }
+      end
+
+      def purchase(amount, credit_card, options = {})
+        # byebug
+        params = {}
+        action = 'purchase'
+        authOnly = false
+        isSettleFunds = true
+        add_bank_amount_purchase(params, amount, authOnly)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_purchase(params, options[:merchant], isSettleFunds, options)
+        commit(action, params, '', '', options)
+      end
+
+      def authorize(amount, credit_card, options = {})
+        #   byebug
+        params = {}
+        action = 'purchase'
+        authOnly = true
+        isSettleFunds = false
+        add_bank_amount_purchase(params, amount, authOnly)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_purchase(params, options[:merchant], isSettleFunds, options)
+        commit(action, params, '', '', options)
+      end
+
+      def refund(amount, credit_card, authCode, options, key_secret)
+        #   byebug
+        params = {}
+        action = 'refund'
+        add_bank_amount_refund(params, amount, authCode)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_refund(params, options)
+        commit(action, params, '', '', key_secret)
+      end
+
+      def capture(amount, authorization, authCode, options = {})
```

Comment:
We expect there to be three headers for `capture` as detailed in the [Contributing](https://github.com/activemerchant/active_merchant/wiki/contributing#generating-the-gateway-skeleton) section. You can pass in the `authCode` in the `options` hash.